### PR TITLE
testing: bybit references, bybit mock testing, pair formatting, standards improvement

### PR DESCRIPTION
**Pull request by gloriousCode**
# PR Description
Just a couple of test improvements to stop some of the errors popping up in Github Actions and the like

Adds Bybit mock tests too

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## How has this been tested

- [x] go test ./... -race
- [x] golangci-lint run
- [x] Coverage of all new functionality

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: currency/pair.go [hunk 1]
```diff
@@ -149,3 +149,11 @@ func MatchPairsWithNoDelimiter(currencyPair string, pairs Pairs, pairFmt PairFor
 	}
 	return EMPTYPAIR, fmt.Errorf("currency %v not found in supplied pairs", currencyPair)
 }
+
+// GetFormatting returns the formatting style of a pair
+func (p Pair) GetFormatting() PairFormat {
+	return PairFormat{
+		Uppercase: p.Base.UpperCase && p.Quote.UpperCase,
```
Comment (shazbert) [inline currency/pair.go:156]: There could be a pair format difference here as well between base and quote 🤷. Just pointing it out don't need to change this.

File: currency/pairs.go [hunk 2]
```diff
@@ -442,3 +443,15 @@ func (p Pairs) ValidateAndConform(pFmt PairFormat, bypassFormatting bool) (Pairs
 	}
 	return formatted, nil
 }
+
+// GetFormatting returns the formatting of a set of pairs
+func (p Pairs) GetFormatting() (PairFormat, error) {
+	if len(p) == 0 {
```
Comment (shazbert) [inline currency/pairs.go:449]: https://github.com/thrasher-corp/gocryptotrader/actions/runs/5874203916/job/15928598358#step:7:140

**Review by shazbert — CHANGES_REQUESTED** [review]
🚀 Just couple of issues. 

```
https://github.com/thrasher-corp/gocryptotrader/actions/runs/5874203916/job/15928598358#step:7:305
```

**Comment by gloriousCode** [comment]
I've added mock testing to bybit as well for the US banning aspect. Happy to revert given v5 PR, but I assume that will take some time and this is nice in the meantime

File: currency/pair.go (hunk 1 above)
Comment (gloriousCode) [reply currency/pair.go:156]: I'll run a test on having an error return for it tomorrow 🎉

**Review by shazbert — APPROVED** [review]
Clickity Clack you get an ACK!

File: exchanges/bybit/bybit_wrapper.go [hunk 3]
```diff
@@ -2126,7 +2126,8 @@ func (by *Bybit) UpdateOrderExecutionLimits(ctx context.Context, a asset.Item) e
 			var pair currency.Pair
 			pair, err = avail.DeriveFrom(pairsData[x].Name)
 			if err != nil {
-				return err
+				log.Warnf(log.ExchangeSys, "unable to load limits for %v, pair data missing", pairsData[x].Name)
```
Comment (thrasher-) [inline exchanges/bybit/bybit_wrapper.go:2129]:
```suggestion
				log.Warnf(log.ExchangeSys, "%s unable to load limits for %v, pair data missing", by.Name, pairsData[x].Name)
```

**Review by thrasher- — APPROVED** [review]
tACK and thanks for adding the bybit mock testing coverage 🚀

**Pull request merged**